phac-nml/irida-next
# Graphql update sample metadata mutation

## What does this PR do and why?
Adding a UpdateSampleMetadata mutation to the GraphQL API which allows a user to update sample metadata.
Fixes #381.

## Screenshots or screen recordings
N/A

## How to set up and validate locally
1. Login as `admin@email.com`.
2. Navigate to `http://localhost:3000/graphiql` to open the graphiql interface .
3. Execute a mutation to update metadata for a Sample.
```graphql
{
  updateSampleMetadata(input: {sampleId: $sample_id, metadata: $metadata}) {
    sample{
      id,
      name,
      description,
      metadata
    },
    status,
    errors
  }
}
```
4. Verify that you see all the metadata for the sample was updated.

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
